semantic-conventions/src/opentelemetry/semconv/model/utils.py

```diff
@@ -17,15 +17,17 @@
 
 from opentelemetry.semconv.model.exceptions import ValidationError
 
-
 ID_RE = re.compile("([a-z](\\.?[a-z0-9_-]+)+)")
+"""Identifiers must start with a lowercase ASCII letter and
+contain only lowercase, digits 0-9, underscore and dots.
```

Comment:
Oh, you are right. That's curious. @thisthat Do you remember the reason for that?
Anyway, `to_const_name`handles that by replacing with underscores so it shouldn't be a big deal.

```python
def to_const_name(name: str) -> str:
    return name.upper().replace(".", "_").replace("-", "_")
```

Reply:
It may make sense if we use `-` for HTTP headers and want to call out specific headers... Although the current PR for that was changed to normalize to `_`, see https://github.com/open-telemetry/opentelemetry-specification/pull/1898

Reply:
Anyway, not really related to this PR, so I'm resolving this conversation.